src/main/scala/com/cognite/sdk/scala/common/dataTypes.scala

```diff
@@ -255,11 +262,44 @@ object NonNullableSetter {
 
   def fromAny[T](value: T): NonNullableSetter[T] = SetValue(value)
 
+  @SuppressWarnings(Array("org.wartremover.warts.Null", "scalafix:DisableSyntax.null"))
   implicit def encodeNonNullableSetter[T](
       implicit encodeT: Encoder[T]
   ): Encoder[NonNullableSetter[T]] = new Encoder[NonNullableSetter[T]] {
     final def apply(a: NonNullableSetter[T]): Json = a match {
       case SetValue(value) => Json.obj(("set", encodeT.apply(value)))
+      // The following cases are covered separately
+      case AddRemoveMap(_, _) => null // scalastyle:ignore null
+      case AddRemoveArr(_, _) => null // scalastyle:ignore null
```

Comment:
I don't know that much about scala. Is this the right way to do this? What is actually the intent of this?

Reply:
It gives some warnings that fail the tests, you need to cover all cases. I put them as placeholders but would like to hear from @exyi or @hakontro on this.

Reply:
The additional encoders I defined, cannot be handled here: 
```
      case AddRemoveMap(add: Seq[A], remove: Seq[A]) =>
        Json.obj(
          ("add" -> Json.arr(add.map(encodeT.apply))),
            ("remove" -> Json.arr(remove.map(encodeT.apply)))
        )
```
the above solution gives`abstract type A in type pattern Seq[A] (the underlying of Seq[A]) is unchecked since it is eliminated by erasure`, I think it means we need to use reflection for type check if we need to handle them in one place 🤔

Reply:
> abstract type A in type pattern Seq[A] (the underlying of Seq[A]) is unchecked since it is eliminated by erasure

this is only a warning, you can suppress it by using `case AddRemoveMap(add: Seq[A] @unchecked, remove: Seq[A] @unchecked)` - https://stackoverflow.com/a/29987749/3577667. Not ideal, but best we can hope for with the erasing generic types...